### PR TITLE
Preload schedules for predictions

### PR DESCRIPTION
**Pull request by lboyarsky**
Ticket: [API: investigate slow prediction queries](https://app.asana.com/0/810933294009540/1126090013342360/f)

Before vs after:

```bash
Levs-MBP:~ lev$ for i in {1..10}; do curl -g "http://localhost:4000/predictions?filter[route]=Red,Orange,Green-B,Green-C,Green-D,Green-E,Blue,Mattapan,741,742,743,751,749&include=schedule,stop,route,trip,vehicle&sort=arrival_time" -s -o /dev/null -w  "%{time_starttransfer}\n"; done
1.822116
1.578963
1.574983
1.562348
1.587852
1.547837
1.458274
1.505999
1.518844
1.454921
Levs-MBP:~ lev$ for i in {1..10}; do curl -g "http://localhost:4000/predictions?filter[route]=Red,Orange,Green-B,Green-C,Green-D,Green-E,Blue,Mattapan,741,742,743,751,749&include=schedule,stop,route,trip,vehicle&sort=arrival_time" -s -o /dev/null -w  "%{time_starttransfer}\n"; done
0.975045
0.920816
0.891342
0.893025
0.931932
0.919227
0.851552
0.872766
0.868732
0.863222
```

### CODE REVIEW

File: apps/api_web/lib/api_web/views/prediction_view.ex [hunk 1]
```diff
@@ -15,6 +15,19 @@ defmodule ApiWeb.PredictionView do
...
     predictions = super(predictions, conn, include_opts)
 
+    predictions =
+      if include_opts != nil and Keyword.has_key?(include_opts, :schedule) do
+        schedules = State.Schedule.schedule_for_many(predictions)
+
+        for p <- predictions,
+            s = Map.get(schedules, {p.stop_id, p.trip_id, p.stop_sequence}),
+            s != nil do
+          Map.put(p, :schedule, s)
```
[the remark this answers is not in the document]
Comment (lboyarsky) [reply apps/api_web/lib/api_web/views/prediction_view.ex]: 1) Fixed.
2) Can you explain why? I don't get this part. Aren't schedules different depending on the stop/trip? 
3) Sure, but I would like to clarify previous item first.

[the remark this answers is not in the document]
Comment (paulswartz) [reply apps/api_web/lib/api_web/views/prediction_view.ex]: Per standup discussion, you only need `{trip_id, stop_sequence}` as the key.

[the remark this answers is not in the document]
Comment (lboyarsky) [reply apps/api_web/lib/api_web/views/prediction_view.ex]: Done

**Comment by lboyarsky** [comment]
Performance impact for schedules, which include predictions, is not that big.. around 10% may be. 

Base version (no changes):

```bash
Levs-MBP:api-lev lev$ for i in {1..10}; do curl -g "http://localhost:4000/schedules?flter[date]=2019-06-18&filter[max_time]=10:44&filter[min_time]=7:44&filter[route]=22,44,23,34,45,14,36,15,38,28,19,41,42,10,32&filter[stop]=15661,15660,1583,1341,1582,1340,390,1581,1580,392,393,1181,394,395,1180,1338,396,1579,1337,1578,1335,1577,399,1334,1576,1498,1575,1179,1332,1497,1574,1331,1573,15820,13321,1339,1351,1350,1346,1345,401,1586,1344,1585,568,1584,1342,404,405,21151,406,407,40001,410,411,412,1330,1572,1571,1570,1174,1173,1172,1569,386,1567,387,1566,388,1485,1328&include=route,trip,stop,prediction,vehicle&api_key=1341e103b7fa4fb58dae4248c6899acf" -s -o /dev/null -w  "%{time_starttransfer}\n"; done
1.499539
1.439951
1.502186
1.433624
1.371786
1.370596
1.219362
1.293876
1.352510
1.329297
```

With code from this PR:
```bash
Levs-MBP:api-lev lev$ for i in {1..10}; do curl -g "http://localhost:4000/schedules?flter[date]=2019-06-18&filter[max_time]=10:44&filter[min_time]=7:44&filter[route]=22,44,23,34,45,14,36,15,38,28,19,41,42,10,32&filter[stop]=15661,15660,1583,1341,1582,1340,390,1581,1580,392,393,1181,394,395,1180,1338,396,1579,1337,1578,1335,1577,399,1334,1576,1498,1575,1179,1332,1497,1574,1331,1573,15820,13321,1339,1351,1350,1346,1345,401,1586,1344,1585,568,1584,1342,404,405,21151,406,407,40001,410,411,412,1330,1572,1571,1570,1174,1173,1172,1569,386,1567,387,1566,388,1485,1328&include=route,trip,stop,prediction,vehicle&api_key=1341e103b7fa4fb58dae4248c6899acf" -s -o /dev/null -w  "%{time_starttransfer}\n"; done
1.340519
1.285699
1.265594
1.276850
1.247792
1.281997
1.183610
1.240096
1.304004
1.283467
```

With merged code from both PRs:
```bash
Levs-MBP:api-lev lev$ for i in {1..10}; do curl -g "http://localhost:4000/schedules?flter[date]=2019-06-18&filter[max_time]=10:44&filter[min_time]=7:44&filter[route]=22,44,23,34,45,14,36,15,38,28,19,41,42,10,32&filter[stop]=15661,15660,1583,1341,1582,1340,390,1581,1580,392,393,1181,394,395,1180,1338,396,1579,1337,1578,1335,1577,399,1334,1576,1498,1575,1179,1332,1497,1574,1331,1573,15820,13321,1339,1351,1350,1346,1345,401,1586,1344,1585,568,1584,1342,404,405,21151,406,407,40001,410,411,412,1330,1572,1571,1570,1174,1173,1172,1569,386,1567,387,1566,388,1485,1328&include=route,trip,stop,prediction,vehicle&api_key=1341e103b7fa4fb58dae4248c6899acf" -s -o /dev/null -w  "%{time_starttransfer}\n"; done
0.879175
0.842114
0.824995
0.858233
0.784421
0.831165
0.798703
0.823349
0.820072
0.870873
```
So yes, I think it would still be helpful to have https://github.com/mbta/api/pull/73.

**Pull request merged**